### ISSUE TITLE
n_process does not have an impact on speeding up processing

### ISSUE THREAD

**Issue by ahmedbesbes-servier** [explosion/spaCy#11321]
Hello everyone, 

I recently started looking into speeding up computations in Spacy and I discovered the `n_process` attribute of the `pipe` method.

However, after trying out multiple values of `n_process` with different sizes of inputs texts, I didn't notice a significant speed-up.

Here's a small snippet I used to check the speed.

```python
import time
from argparse import ArgumentParser
import spacy
from tqdm import tqdm

nlp = spacy.load("en_core_web_sm", disable=["ner"])


if __name__ == "__main__":
    parser = ArgumentParser()
    parser.add_argument("--num", type=int, default=1000)
    parser.add_argument("--n-process", type=int, default=1)
    parser.add_argument("--bs", type=int, default=50)

    args = parser.parse_args()

    print(f"num documents: {args.num}")
    print(f"num processes: {args.n_process}")
    print(f"batch size: {args.bs}")

    texts = ["this protein impacts this molecule very easily"] * args.num

    t0 = time.time()
    docs = []
    for doc in tqdm(
        nlp.pipe(
            texts,
            n_process=args.n_process,
            batch_size=args.bs,
            disable=["ner"],
        ),
        total=len(texts),
    ):
        docs.append(doc)

    t1 = time.time()

    print(f"processing {len(docs)} documents took {t1 - t0} s")

```

Here's a screenshot of some tests:

<img width="716" alt="Capture d’écran 2022-08-16 à 17 59 07" src="https://user-images.githubusercontent.com/107037348/184925374-37eb69b9-7fd3-4936-ba37-548e3c551652.png">

Am I doing something wrong?

Thanks for your help,

## Your Environment

* Operating System: Mac OS Big Sur 11.6
* Python Version Used: 3.9.12
* spaCy Version Used: 3.2.4
* Environment Information:

**Comment by shadeMe** [comment]
Could you upgrade to the latest version of spaCy and re-run your benchmarks? We fixed a performance regression in v3.3.0 that affected the `pipe` API.

**Comment by ahmedbesbes-servier** [comment]
@shadeMe thanks! it seems to be working now 😄
